src/predis/mock.clj

```diff
@@ -287,6 +290,22 @@
       (when (<= idx last-idx)
         (get vs idx'))))
 
+  (core/linsert [this k pos pivot v]
+    (let [vs (vec (core/get this k))
+          pos' (string/lower-case pos)]
+      (assert (#{"before" "after"} pos') err-syntax)
+      (if (seq vs)
+        (if (contains? (set vs) pivot)
+          (let [[before after]
+                (split-at (if (= pos' "before")
+                            (.indexOf vs pivot)
+                            (inc (.indexOf vs pivot))) vs)
```

Comment:
One thing that might make this cleaner is extracting `split-idx`, that way you can just say

```clj
[before after] (split-at split-idx vs)
```